### PR TITLE
Scaffolding for plugin state files

### PR DESCRIPTION
**Pull request by chinchiheather**
- New files: `actions.ts`, `reducer.ts`, `pm-plugins/plugin-factory.ts`
- Updated existing files to add some extra content when using plugin state: `commands.ts`, `types.ts`, `pm-plugins-main.ts`
- Restructured templates folder to be able to have a folder just for plugin state files, put templates inside `files/templates` and added new folder `files/plugin-state-templates` for the new ones

### CODE REVIEW

File: src/twp-editor-plugin/files/plugin-state-templates/reducer.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { <%= classify(name) %>PluginState } from './types';
+import { <%= classify(name) %>ActionTypes, <%= classify(name) %>Action } from './actions';
+
+export function reducer(state: <%= classify(name) %>PluginState, action: <%= classify(name) %>Action): <%= classify(name) %>PluginState {
+  switch(action.type) {
```
Comment (timhor) [inline src/twp-editor-plugin/files/plugin-state-templates/reducer.ts:5]: Should we add the sample action here as well?
```
case <%= classify(name) %>ActionTypes.SOME_ACTION:
  return { ...state, /* fill this in */ };
```

Comment (chinchiheather) [reply src/twp-editor-plugin/files/plugin-state-templates/reducer.ts:5]: Yeah good idea

File: src/twp-editor-plugin/files/plugin-state-templates/pm-plugins/plugin-factory.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { pluginFactory } from '../../utils/plugin-state-factory';
+import { <%= camelize(name) %>PluginKey } from '../plugin-key';
+import { <%= classify(name) %>PluginState } from '../types';
+import { reducer } from '../reducer';
+
+export const {
+  createCommand,
+  getPluginState,
+  createPluginState,
+} = pluginFactory(<%= camelize(name) %>PluginKey, reducer);
```
Comment (timhor) [inline src/twp-editor-plugin/files/plugin-state-templates/pm-plugins/plugin-factory.ts]: Would it be helpful to add a placeholder line
```
/* { mapping, onDocChanged, onSelectionChanged }, */
```
at the end?

File: src/twp-editor-plugin/files/plugin-state-templates/pm-plugins/plugin-factory.ts [hunk 3]
```diff
@@ -7,4 +7,4 @@ export const {
   createCommand,
   getPluginState,
   createPluginState,
-} = pluginFactory(<%= camelize(name) %>PluginKey, reducer);
+} = pluginFactory(<%= camelize(name) %>PluginKey, reducer /**, { mapping, onDocChanged, onSelectionChanged } **/);
```
Comment (timhor) [inline src/twp-editor-plugin/files/plugin-state-templates/pm-plugins/plugin-factory.ts:10]: _Tiny_ change, but I think it'd be better if the trailing comma was included, as Prettier would add it anyway inside the AF repo

**Review by timhor — APPROVED** [review]

**Pull request merged**